src/execution/futex-emulation.cc

```diff
@@ -543,6 +559,9 @@ Object FutexEmulation::WaitSync(Isolate* isolate,
   return *result;
 }
 
+FutexWaitListNode::FutexWaitListNode() {
+  recordreplay::RegisterPointer("FutexWaitListNode", this);
```

Comment:
I believe Brian recommended moving toward the `RecordReplayId()` method which we have `HAS_RECORD_REPLAY_ID` and `INIT_RECORD_REPLAY_ID` macros for

Reply:
I assume you mean the `NewIdMainThread()` and `NewIdAnyThread()` mechanism.  I considered doing that, but there are no uses of those within the v8 codebase - those mechanisms are defined in chromium, and don't seem directly referencable from v8.

I'd have to basically port the mechanism over to the v8 codebase and then use it internally, but I don't think it's worth it.  These asserts should be removed shortly once we confirm that this bug is fixed.  It should be fine to use the old `pointerId` mechanism on an ephemeral basis.  @bhackett1024 what do you think?